docs/configuration/fields.md

```diff
@@ -0,0 +1,301 @@
+
+Each Source is defined by the property ``` "type" ```. Allowed values for each type of source are:
+
+1. ["static"](/cf-nodejs-logging-support/configuration/fields#static)
+1. ["env"](/cf-nodejs-logging-support/configuration/fields#env)
+1. ["config-field"](/cf-nodejs-logging-support/configuration/fields#config-field)
+1. ["req-header"](/cf-nodejs-logging-support/configuration/fields#req-header)
+1. ["req-body"](/cf-nodejs-logging-support/configuration/fields#req-body)
+1. ["res-header"](/cf-nodejs-logging-support/configuration/fields#res-header)
+1. ["res-body"](/cf-nodejs-logging-support/configuration/fields#res-body)
+1. ["uuid"](/cf-nodejs-logging-support/configuration/fields#uuid)
+
+To get more information about each type of source go to [source types](/cf-nodejs-logging-support/configuration/fields#source-types).
+
+To get more information about assigning multiple sources to a field go to [multiple sources](/cf-nodejs-logging-support/configuration/fields#source-types).
+
+### 3. output (required)
+
+  * **Type:** ```Array```
+  * **Description:** Define output of field (message logs, request logs or both).
+  * Allowed values:
+    * ["msg-log"] (Only log in [message logs](/cf-nodejs-logging-support/general-usage/message-logs))
+    * ["req-log"] (Only log in [request logs](/cf-nodejs-logging-support/general-usage/request-logs))
+    * ["msg-log","req-log"] (Log in both [message](/cf-nodejs-logging-support/general-usage/message-logs) and [request logs](/cf-nodejs-logging-support/general-usage/request-logs) )
+
+### 4. disable (optional)
+
+  * **Type:** ```boolean```
+  * **Description:** If ```true```, ommit field.
+
+### 5. default (optional)
+
+  * **Type:** ```string```
+  * **Description:** If returned value from source is ```null```, then assign this value.
+
+---
+
+## Source types
+
+### 1. ``` "static" ```
+
+Return always a static value. Declare this value as string in the property ```"value"```.
+
+Example:
+
+  ```js
+    {
+        "type": "static",
+        "value": "my_value",
+        "output": ["msg-log", "req-log"]
+    }
+  ```
+
+### 2. ``` "env" ```
+
+Read value from environment variable. Declare the environment variable as string in the property ```"name"```.
+
+Example:
+
+```js
+    {
+        "type": "env",
+        "path": "ENV_VAR",
+        "output": ["msg-log"]
+    }
+```
+
+If you need to access a subproperty of the environmet variable, declare the property ```"path"``` with the path as an array of strings.
+
+Example:
+
+```js
+    {
+        "type": "env",
+        "path": ["ENV_VAR", "application_id"],
+        "output": ["msg-log"]
+    }
+```
+
+### 3. ``` "config-field" ```
+
+Copy value from another configured field. Declare the name of the field to copy in the property ```"name"```.
+
+Example:
+
+  ```js
+    {
+        "type": "config-field",
+        "name": "field-a",
+        "output": ["msg-log"]
+    }
+  ```
```

Comment:
I think there is a misunderstanding, that list shows examples of `Source` objects, that means the objects assigned in the property `"source": {...}`.

The property  that represent the output log name is in a higher level.

Example ConfigField:
```
{
    "name": "new-field",
    ...
    "source": 
        {
            "type": "config-field",
            "name": "field-a",
            ...
        },
    ...
}
```

To avoid this confusion I would suggest either writing the source examples as this for every source type or changing the source property `name` to `"from"` or `"field-name"`. What do you think?